DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/EncryptionProcessor.cs

```diff
@@ -0,0 +1,100 @@
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Dicom.DeID.SharedLib;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors
+{
+    public class EncryptionProcessor : IAnonymizerProcessor
+    {
+        private readonly EncryptFunction _encryptFunction;
+
+        public EncryptionProcessor(JObject settingObject, IAnonymizerSettingsFactory settingFactory = null)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            settingFactory ??= new AnonymizerSettingsFactory();
+            var encryptionSetting = settingFactory.CreateAnonymizerSetting<EncryptionSetting>(settingObject);
+            _encryptFunction = new EncryptFunction(encryptionSetting);
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context = null)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            try
+            {
+                if (item is DicomStringElement)
+                {
+                    var encryptedValues = ((DicomStringElement)item).Get<string[]>().Where(x => !string.IsNullOrEmpty(x)).Select(x => EncryptToBase64String(x));
+                    if (encryptedValues.Count() != 0)
+                    {
+                        dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, encryptedValues.ToArray());
+                    }
+                }
+                else if (item is DicomOtherByte)
+                {
+                    var valueBytes = ((DicomOtherByte)item).Get<byte[]>();
+                    var encryptesBytes = _encryptFunction.EncryptContentWithAES(valueBytes);
+                    dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, encryptesBytes);
+                }
+                else if (item is DicomFragmentSequence)
+                {
+                    var enumerator = ((DicomFragmentSequence)item).GetEnumerator();
+
+                    var element = item.ValueRepresentation == DicomVR.OW
+                        ? (DicomFragmentSequence)new DicomOtherWordFragment(item.Tag)
+                        : new DicomOtherByteFragment(item.Tag);
+
+                    while (enumerator.MoveNext())
+                    {
+                        element.Fragments.Add(new MemoryByteBuffer(_encryptFunction.EncryptContentWithAES(enumerator.Current.Data)));
+                    }
+
+                    dicomDataset.AddOrUpdate(element);
+                }
+                else
+                {
+                    throw new AnonymizationOperationException(DicomAnonymizationErrorCode.UnsupportedAnonymizationFunction, $"Encrypt is not supported for {item.ValueRepresentation}");
+                }
+            }
+            catch (Exception ex)
```

Comment:
explicitly catch (DicomValidationException) then throw a new AnonymizationOperationException
No need to throw other exceptions any more.

Reply:
Thanks, fixed it.